source/string.c

```diff
@@ -4,6 +4,123 @@
  */
 #include <aws/common/string.h>
 
+#ifdef _WIN32
+#    include <windows.h>
+
+struct aws_string *aws_string_convert_to_wchar_str(
+    struct aws_allocator *allocator,
+    const struct aws_string *to_convert) {
+    struct aws_byte_cursor convert_cur = aws_byte_cursor_from_string(to_convert);
+    return aws_string_convert_to_wchar_from_byte_cursor(allocator, &convert_cur);
+}
+
+struct aws_string *aws_string_convert_to_wchar_from_byte_cursor(
+    struct aws_allocator *allocator,
+    const struct aws_byte_cursor *to_convert) {
+    /* if a length is passed for the to_convert string, converted size does not include the null terminator,
+     * which is a good thing. */
+    int converted_size = MultiByteToWideChar(CP_UTF8, 0, (const char *)to_convert->ptr, (int)to_convert->len, NULL, 0);
```

Comment:
Accessing `to_convert->` without checking for nullness could lead to null-pointer dereferences.